### PR TITLE
Add spacing around event handlers when converting ts->py; add support for whitespace sensitive test cases

### PR DESCRIPTION
**Pull request by darzu**
Fixes: https://github.com/microsoft/pxt-microbit/issues/3043#event-3333365451

These blocks:
image

Before:
image

After:
image

Also adds test cases and support for whitespace sensitive testing for the py->ts and ts->py syntax tests.
This also adds infrastructure for more easily specifying per-case test options (like whitespace sensitivity).

### CODE REVIEW

File: pxtpy/pydecompiler.ts [hunk 1]
```diff
@@ -674,11 +682,16 @@ namespace pxt.py {
...
 
-            let [exp, expSup] = emitExp(s.expression)
-            return expSup.concat(exp)
+            const [exp, expSup] = emitExp(s.expression)
+            if (expSup.length)
+                // If an expression has supporting statements, this is usually an event handler
+                // and we usually want it to be padded with empty lines.
+                return ["", ...expSup, ...exp, ""]
+            else
+                return [...expSup, ...exp]
```
Comment (riknoll) [inline pxtpy/pydecompiler.ts:691]: Nitpick: no need for `...expSup`

Comment (darzu) [reply pxtpy/pydecompiler.ts:691]: Ha, good catch
